### PR TITLE
Adding unit tests for SetupDataPkg

### PR DESCRIPTION
**Pull request by kuqin12**
This change adds unit tests for SetupDataPkg.

A few minor issues are also fixed during unit test discovery.

### CODE REVIEW

**Review by cfernald — APPROVED** [review]

File: SetupDataPkg/ConfDataSettingProvider/UnitTest/ConfDataSettingProviderUnitTest.inf [hunk 1]
```diff
@@ -0,0 +1,53 @@
+## @file
+# Unit tests of the DxeResetSystemLib instance of the ResetSystemLib class
```
[the remark this answers is not in the document]
Comment (kuqin12) [reply SetupDataPkg/ConfDataSettingProvider/UnitTest/ConfDataSettingProviderUnitTest.inf]: Updated.

File: SetupDataPkg/Library/ConfigBlobBaseLib/UnitTest/ConfigBlobBaseLibUnitTest.c [hunk 2]
```diff
@@ -0,0 +1,490 @@
+/** @file
+  Unit tests of the DxeResetSystemLib instance of the ResetSystemLib class
```
[the remark this answers is not in the document]
Comment (kuqin12) [reply SetupDataPkg/Library/ConfigBlobBaseLib/UnitTest/ConfigBlobBaseLibUnitTest.c]: Updated.

File: SetupDataPkg/Test/MockLibrary/MockResetSystemLib/MockResetSystemLib.c [hunk 3]
```diff
@@ -0,0 +1,40 @@
+/** @file
+  Reset System Library functions for PCAT platforms
```
[the remark this answers is not in the document]
Comment (kuqin12) [reply SetupDataPkg/Test/MockLibrary/MockResetSystemLib/MockResetSystemLib.c]: Yeah, it was reused from edk2 ResetSystemLib which has this header. It is updated now

File: SetupDataPkg/ConfDataSettingProvider/UnitTest/ConfDataSettingProviderUnitTest.h [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+  OUT       UINT8                  *Value
+  );
+
+/**
+  Set new configuration value to variable storage.
+
+  @param This      Provider Setting
+  @param Value     a pointer to a datatype defined by the Type for this setting.
+  @param ValueSize Size of the data for this setting.
+  @param Flags     Informational Flags passed to the SET and/or Returned as a result of the set
```
Comment (kenlautner) [inline SetupDataPkg/ConfDataSettingProvider/UnitTest/ConfDataSettingProviderUnitTest.h:122]: This parameter isn't included in the actual function.

File: SetupDataPkg/Library/ConfigDataLib/UnitTest/ConfigDataLibUnitTest.c [hunk 5]
```diff
@@ -0,0 +1,327 @@
...
+
+#include <Good_Config_Data.h>
+
+#define UNIT_TEST_APP_NAME     "Conf Data Iterator Unit Tests"
+#define UNIT_TEST_APP_VERSION  "1.0"
+
+/**
+  Handler function dispatched for individual tag based data.
+
+  @param[in] ConfDataPtr  Pointer to configuration data.
```
Comment (kenlautner) [inline SetupDataPkg/Library/ConfigDataLib/UnitTest/ConfigDataLibUnitTest.c:31]: The parameter here isn't any of the ones listed in the actual function.  Also none of the actual input parameters are described.

File: pip-requirements.txt [hunk 6]
```diff
@@ -12,7 +12,7 @@
 # https://www.python.org/dev/peps/pep-0440/#version-specifiers
 ##
 
-edk2-pytool-library~=0.11.0 # MU_CHANGE - update to 0.11.0 or later
-edk2-pytool-extensions~=0.17.0 # MU_CHANGE - update to 0.15.0 or later
+edk2-pytool-library~=0.11.2 # MU_CHANGE - update to 0.11.0 or later
+edk2-pytool-extensions~=0.17.1 # MU_CHANGE - update to 0.15.0 or later
```
Comment (kenlautner) [inline pip-requirements.txt:16]: Maybe update the comments to match the newer versions.  I don't believe everything has been backwards compatible.

File: SetupDataPkg/ConfDataSettingProvider/UnitTest/ConfDataSettingProviderUnitTest.c [hunk 7]
```diff
@@ -0,0 +1,1955 @@
...
+  If ProtocolGuid is NULL, then ASSERT().
+  If NotifyTpl is not a legal TPL value, then ASSERT().
+  If NotifyFunction is NULL, then ASSERT().
+  If Registration is NULL, then ASSERT().
+
+
+  @param  ProtocolGuid    Supplies GUID of the protocol upon whose installation the event is fired.
+  @param  NotifyTpl       Supplies the task priority level of the event notifications.
+  @param  NotifyFunction  Supplies the function to notify when the event is signaled.
+  @param  NotifyContext   The context parameter to pass to NotifyFunction.
```
Comment (kenlautner) [inline SetupDataPkg/ConfDataSettingProvider/UnitTest/ConfDataSettingProviderUnitTest.c:188]: Maybe include the optional tag here as well.

File: SetupDataPkg/ConfDataSettingProvider/UnitTest/ConfDataSettingProviderUnitTest.c [hunk 8]
```diff
@@ -0,0 +1,1955 @@
...
+  .SetVariable = MockSetVariable,
+};
+
+EFI_STATUS
+EFIAPI
+MockLocateProtocol (
+  IN  EFI_GUID *Protocol,
+  IN  VOID *Registration, OPTIONAL
+  OUT VOID      **Interface
+  )
```
Comment (kenlautner) [inline SetupDataPkg/ConfDataSettingProvider/UnitTest/ConfDataSettingProviderUnitTest.c:392]: Any reason why there isn't a function description for this function?

File: SetupDataPkg/ConfApp/UnitTest/ConfAppBootOptionUnitTest.c [hunk 9]
```diff
@@ -0,0 +1,579 @@
...
+  return Ret;
+}
+
+EFI_STATUS
+EFIAPI
+MockWaitForEvent (
+  IN  UINTN      NumberOfEvents,
+  IN  EFI_EVENT  *Event,
+  OUT UINTN      *Index
+  )
```
Comment (kenlautner) [inline SetupDataPkg/ConfApp/UnitTest/ConfAppBootOptionUnitTest.c:194]: No function description.

File: SetupDataPkg/ConfApp/UnitTest/ConfAppSetupConfUnitTest.c [hunk 10]
```diff
@@ -0,0 +1,890 @@
...
+};
+
+/*
+Set a single setting
+
+@param[in] This:       Access Protocol
+@param[in] Id:         Setting ID to set
+@param[in] AuthToken:  A valid auth token to apply the setting using.  This auth token will be validated
+                       to check permissions for changing the setting.
+@param[in] Type:       Type that caller expects this setting to be.
```
Comment (kenlautner) [inline SetupDataPkg/ConfApp/UnitTest/ConfAppSetupConfUnitTest.c:75]: Missing the ValueSize parameter.

**Review by kenlautner — APPROVED** [review]
See comments on some changes.

File: SetupDataPkg/ConfDataSettingProvider/UnitTest/ConfDataSettingProviderUnitTest.h (hunk 4 above)
Comment (kuqin12) [reply SetupDataPkg/ConfDataSettingProvider/UnitTest/ConfDataSettingProviderUnitTest.h:122]: Updated.

File: SetupDataPkg/ConfDataSettingProvider/UnitTest/ConfDataSettingProviderUnitTest.c (hunk 8 above)
Comment (kuqin12) [reply SetupDataPkg/ConfDataSettingProvider/UnitTest/ConfDataSettingProviderUnitTest.c:376]: Updated.

File: SetupDataPkg/ConfApp/UnitTest/ConfAppBootOptionUnitTest.c (hunk 9 above)
Comment (kuqin12) [reply SetupDataPkg/ConfApp/UnitTest/ConfAppBootOptionUnitTest.c:196]: Updated.

File: SetupDataPkg/Library/ConfigDataLib/UnitTest/ConfigDataLibUnitTest.c (hunk 5 above)
Comment (kuqin12) [reply SetupDataPkg/Library/ConfigDataLib/UnitTest/ConfigDataLibUnitTest.c]: Thanks, updated.

File: pip-requirements.txt (hunk 6 above)
Comment (kuqin12) [reply pip-requirements.txt]: Updated.

File: SetupDataPkg/ConfApp/UnitTest/ConfAppSetupConfUnitTest.c (hunk 10 above)
Comment (kuqin12) [reply SetupDataPkg/ConfApp/UnitTest/ConfAppSetupConfUnitTest.c]: Parameter added

File: SetupDataPkg/ConfDataSettingProvider/UnitTest/ConfDataSettingProviderUnitTest.c (hunk 7 above)
Comment (kuqin12) [reply SetupDataPkg/ConfDataSettingProvider/UnitTest/ConfDataSettingProviderUnitTest.c]: Udpated.

**Pull request merged**